### PR TITLE
wireless-regdb: update to 2019-06-03

### PR DESCRIPTION
**Pull request by zacharyrs**
Fixes build issues on a python3 host (issues with the print statement
formatting in the current build)

Includes 100-regdb-write-firmware-file-format-version-code-20.patch and
other fixes.

Closes https://bugs.openwrt.org/index.php?do=details&task_id=1605.

Signed-off-by: Zachary Riedlshah <git@zacharyrs.me>

### CODE REVIEW

**Comment by diizzyy** [comment]
You need python-future installed

**Comment by diizzyy** [comment]
Use the URL instead of git

File: package/firmware/wireless-regdb/Makefile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-PKG_SOURCE_PROTO:=git
-PKG_SOURCE_URL:=https://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.git
-PKG_SOURCE_DATE:=2017-10-20
-PKG_SOURCE_VERSION:=4343d359ed5e7404de8803a74df186457b26ab79
-PKG_MIRROR_HASH:=5f5b669f32ae36cb65b1d99efbbbfd42c2983cda32f6448346e3e54ffaba3889
+PKG_SOURCE:=wireless-regdb-master-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.git/snapshot/
+PKG_HASH:=6737a742cbd4c9740171a537faff0b3ec2813632d74efb09bccfe39e605d7fb6
+PKG_CAT:=zcat
```
Comment (zacharyrs) [reply package/firmware/wireless-regdb/Makefile]: Probably not.

I added it as it was in the wiki example - this is the first time I've touched openwrt.

Removed it now.

**Comment by zacharyrs** [comment]
Updated as requested, and squashed with previous commits.

File: package/firmware/wireless-regdb/Makefile (hunk 1 above)
Comment (diizzyy) [inline package/firmware/wireless-regdb/Makefile]: Really needed?

Comment (zacharyrs) [reply package/firmware/wireless-regdb/Makefile]: Apologies, is what really needed?

Comment (diizzyy) [reply package/firmware/wireless-regdb/Makefile]: ```PKG_CAT```

**Comment by zacharyrs** [comment]
I had to make a number of changes as I misunderstood how to format the Makefile.

The current version has a mix of spaces and tabs, so I swapped it all to spaces, which broke it. Now swapped back to tabs, which works.

Also had to define the PKG_BUILD_DIR, as the snapshot link includes 'master'.

Compiling fine locally now.

**Comment by zacharyrs** [comment]
Apologies if you're busy, but I just thought I'd follow up regarding this...

**Comment by robimarko** [comment]
@zacharyrs Thanks for the work, I was thinking about doing something similar but you beat me to it.
I also needed updated regdb

**Comment by zacharyrs** [comment]
@robimarko Thanks, hopefully it'll get merged soon!

**Comment by nasbdh9** [comment]
If not run `pip install future` 
there will be an error `ImportError: No module named builtins`
Is this a problem with Python 2.x?

**Comment by dengqf6** [comment]
@diizzyy 
There are release tarballs in kernel.org mirror
`@KERNEL/software/network/wireless-regdb/wireless-regdb-2018.10.24.tar.xz`
Please use it instead

**Comment by diizzyy** [comment]
If we have release tarballs by all means, go for it.

**Comment by diizzyy** [comment]
@nasbdh9 
https://python-future.org/quickstart.html#installation

**Comment by dengqf6** [comment]
A newer version is available
`wireless-regdb-2019.03.01.tar.xz`

**Comment by zacharyrs** [comment]
Sorry, never stuck this on my todo list so it sat there...
Looks like it builds fine!

Done now!

**Comment by nbd168** [comment]
Fails on my host (with python2) with the following error:

python /Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/db2fw.py /Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/regulatory.db /Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/db.txt
Traceback (most recent call last):
  File "/Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/db2fw.py", line 6, in <module>
    from dbparse import DBParser
  File "/Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/dbparse.py", line 3, in <module>
    from builtins import bytes
ImportError: No module named builtins

This will only build if the host has the "future" module installed, which we don't currently rely on.
I would like to avoid this extra host dependency if possible.

**Comment by zacharyrs** [comment]
The future dependency is introduced in the 2018.05.09 version, which I think is when the scripts were migrated to be python 3 compatible.

I could create a couple of patches to try revert the changes that rely on futures, but that seems a little counterproductive, considering python 2 is only supported for another 9 months...

**Comment by nbd168** [comment]
Maybe you could make a host package that builds the module and installs it to staging dir, and then add some env vars or patches to make wireless-regdb use the module from there.

**Comment by zacharyrs** [comment]
Unfortunately that sounds a bit out of my expertise...

I'm rather busy with uni and work right now, so this has a reasonably low priority.

I'll likely just manually patch the repo when I rebuild, or possibly create a custom feed, at least until python 3 is targeted.

**Comment by diizzyy** [comment]
Having the OpenSSL bump in mind perhaps it's worth changing the dependency to python3 instead of doing it the last minute? https://pythonclock.org/

**Comment by diizzyy** [comment]
Hmm....
Not sure if it's true for all installs but on my box (Alpine Linux) python2 packages get built before wireless-regdb which also adds a host version of python that doesn't have python-future resulting in a compilation failure.

**Comment by orangepizza** [comment]
and now another new version https://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.git/tag/?h=master-2019-06-03

**Comment by zacharyrs** [comment]
@orangepizza I'll update it this evening.

**Comment by zacharyrs** [comment]
Sorted!

**Comment by dengqf6** [comment]
Why did you not use official release tarball anymore?

**Comment by zacharyrs** [comment]
Woops! I just messed it up and grabbed an old commit.
Will swap it over now.

File: package/firmware/wireless-regdb/Makefile [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
-PKG_SOURCE_PROTO:=git
-PKG_SOURCE_URL:=https://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.git
-PKG_SOURCE_DATE:=2017-10-20
-PKG_SOURCE_VERSION:=4343d359ed5e7404de8803a74df186457b26ab79
-PKG_MIRROR_HASH:=5f5b669f32ae36cb65b1d99efbbbfd42c2983cda32f6448346e3e54ffaba3889
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
+PKG_SOURCE_URL:=@KERNEL/software/network/wireless-regdb/
+PKG_HASH:=cd917ed86b63ce8d93947979f1f18948f03a4ac0ad89ec25227b36ac00dc54bf
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
Comment (dengqf6) [inline package/firmware/wireless-regdb/Makefile]: `PKG_BUILD_DIR` can be removed. It's already the default

**Comment by orangepizza** [comment]
newest patch fails to complie a image:

```
make[3]: Entering directory '/mnt/realdisk/openwrt/openwrt/package/firmware/wireless-regdb'
rm -f /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/.built
touch /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/.built_check
python /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/db2fw.py /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/regulatory.db /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/db.txt
Traceback (most recent call last):
  File "/mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/db2fw.py", line 6, in <module>
    from dbparse import DBParser
  File "/mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/dbparse.py", line 3, in <module>
    from builtins import bytes
ImportError: No module named builtins
make[3]: *** [Makefile:35: /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/.built] Error 1
make[3]: Leaving directory '/mnt/realdisk/openwrt/openwrt/package/firmware/wireless-regdb'
time: package/firmware/wireless-regdb/compile#0.11#0.03#0.13
make[2]: *** [package/Makefile:109: package/firmware/wireless-regdb/compile] Error 2
make[2]: Leaving directory '/mnt/realdisk/openwrt/openwrt'
make[1]: *** [package/Makefile:103: /mnt/realdisk/openwrt/openwrt/staging_dir/target-mipsel_24kc_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/mnt/realdisk/openwrt/openwrt'
make: *** [/mnt/realdisk/openwrt/openwrt/include/toplevel.mk:218: world] Error 2

```

**Comment by orangepizza** [comment]
is there a way to force buildroot to use python3?

**Comment by ynezz** [comment]
> is there a way to force buildroot to use python3?

Nope, see #1937

File: package/firmware/wireless-regdb/Makefile (hunk 2 above)
Comment (zacharyrs) [reply package/firmware/wireless-regdb/Makefile]: Fixed!

**Comment by zacharyrs** [comment]
@orangepizza As per the comment above, you need `future` installed.

**Comment by ynezz** [comment]
@zacharyrs Thanks a lot for your patience and continuous support! I've merged this into my staging tree, where I'm finalizing the Python3 support and I'm not able to build test it without this PR.

**Comment by dengqf6** [comment]
You can add `PKGARCH:=ALL` to `define Package/wireless-regdb` section, as it's an architecture-independent database file

**Pull request closed without merging**